src/prefect/utilities/engine.py

```diff
@@ -113,16 +101,12 @@ def add_futures_and_states_to_inputs(obj):
         max_depth=max_depth,
     )
 
-    await asyncio.gather(*[future._wait_for_submission() for future in futures])
-    for future in futures:
-        inputs.add(TaskRunResult(id=future.task_run.id))
-
```

Comment:
The changes here are a guess, based on what happened in 0d89908ea7023894893b9a13b3d23e942777c264. @desertaxle I wonder why this hasn't broken before?

Reply:
Ah, it's probably because this is an old implementation, and we use the one below in the engine now. It looks like there are a couple of places where this is still used, but we should replace those uses with the sync version.

Reply:
I'll leave updating those (2) locations to you then. :-) I will remove `collect_task_run_inputs_sync()` in this module as it is broken too and not used anywhere.

Reply:
Another one that is broken is `resolve_inputs()` in this module, it expects to find futures with a `_wait()` method retuning `State` instances. It is still being used in `prefect.runner.submit()`.